service/torrent/metainfoFetcher/metainfoFetcher.go

```diff
@@ -150,12 +152,28 @@ func (fetcher *MetainfoFetcher) gotResult(r Result) {
 	}
 
 	if !updatedSuccessfully {
-		fetcher.failedOperations[r.operation.torrent.ID] = struct{}{}
+		fetcher.failedOperations[r.operation.torrent.ID] = time.Now()
 	}
 
 	fetcher.removeFromQueue(r.operation)
 }
 
+func (fetcher *MetainfoFetcher) removeOldFailures() {
+	// Cooldown is disabled
+	if fetcher.failCooldown < 0 {
+		return
+	}
+
+	now := time.Now()
+	for id, failTime := range fetcher.failedOperations {
+		if failTime.Add(time.Duration(fetcher.failCooldown) * time.Second).Before(now) {
+			log.Infof("Torrent TID %d gone through cooldown, removing from failures")
+			// Deleting keys inside a loop seems to be safe.
+			delete(fetcher.failedOperations, id)
```

Comment:
I was afraid too, but after some Googling it seems to be safe. [https://github.com/golang/go/issues/9926](url)